Dispatcher/UI/Auth/AuthViewController.swift

```diff
@@ -7,57 +7,196 @@
 
 import UIKit
 
-class AuthViewController: UIViewController, SignupOrLoginDelegate{
-
-    var authViewModel = AuthViewModel()
-    var logoHeader: HeaderView?
-    var signupOrLoginView: SignupOrLoginView?
-
+class AuthViewController: UIViewController{
+    
+    @IBOutlet weak var scrollView: UIScrollView!
+    @IBOutlet var contentView: UIView!
+    @IBOutlet weak var headerView: HeaderView!
+    @IBOutlet weak var authView: UIView!
+    
+    @IBOutlet weak var emailTextField: TextFieldWithValidation!
+    @IBOutlet weak var emailErrorLabel: UILabel!
+    
+    @IBOutlet weak var passwordTextField: TextFieldWithValidation!
+    @IBOutlet weak var passwordErrorLabel: UILabel!
+    
+    @IBOutlet weak var reEnterPasswordTextField: TextFieldWithValidation!
+    @IBOutlet weak var reEnterPasswordErrorLabel: UILabel!
+    
+    @IBOutlet weak var authModeLabel: UILabel!
+    @IBOutlet weak var primaryButton: DispatcherAppButton!
+    @IBOutlet weak var secondaryButton: DispatcherAppButton!
+    
+    var authViewModel = AuthViewModel.shared
+    var authMode: AuthMode = .Signup
+    var validatedEmail: String = ""
+    var validatedPassword: String = ""
+    
     override func viewDidLoad() {
         super.viewDidLoad()
         
-        let screenHeight = UIScreen.main.bounds.height
-        let screenWidth = UIScreen.main.bounds.width
-                
-        logoHeader = HeaderView(frame: CGRect(x: 0, y: 0, width: screenWidth, height: screenHeight * 0.45), headerType: .logoHeader)
-        self.view.addSubview(logoHeader ?? UIView())
+        headerView.initView(headerType: .logoHeader)
+        emailTextField.initTextField(type: .email)
```

Comment:
Create a setupTextFields function and put the init & delegate lines of the textfields inside it then call it inside the viewDidLoad